### PR TITLE
Finer control over when Python tests are run

### PR DESCRIPTION
**Pull request by jignparm**
In build.py, should run python tests only if --enable_pybind or --build-wheel options are enabled, and not --skip_onnx_tests.

Removed --build-wheel from docker file -- this should be passed in via -x argument, not hard coded in the script. Does not run on x86

### CODE REVIEW

**Comment by jywu-msft** [comment]
please make appropriate changes to ensure we don't lose python test coverage.

> Removed --build-wheel from docker file -- this should be passed in via -x argument, not hard coded in the script. Does not run on x86

**Comment by snnn** [comment]
Why we have both ' --skip_onnx_tests' and '--enable_onnx_tests". Should just one of them is enough?

**Comment by jignparm** [comment]
@jywu-msft , updated to add **--build_wheel** option to linux-ci-pipeline.yml instead of docker **run_build.sh**. This will ensure python test coverage in CI, and also not force python tests on x86 builds.

  >Why we have both ' --skip_onnx_tests' and '--enable_onnx_tests". Should just one of them is enough?

@snnn This was already there previously. This PR only changes

**Comment by jywu-msft** [comment]
Unfortunately updating just the Linux CPU CI yaml isn't sufficient. We were testing python bindings and wheel creation for all providers on Linux. ie gpu, tensorrt, ngraph etc
And we have other providers coming up as well.

The idea was that python should be supported for all Linux build flavors. 

How did this work before this change? 
If x86 is only flavor which can't support python,
Alternative option is to keep the build_wheel lit in run docker
Instead, check for x86 build arg and skip python in build.py?

**Comment by jignparm** [comment]
> The idea was that python should be supported for all Linux build flavors

Updated all the ci pipelines with  **--build_wheel.** This should address this concern. The plan was to do this in another pending PR to avoid merge conflicts. But either way is fine.

>How did this work before this change?

x86 builds are not part of CI -- they need to be included, but before we can include them, **run_build.sh** script needs to be fixed so that it doesn't force python tests on x86.

**Comment by jywu-msft** [comment]
> > The idea was that python should be supported for all Linux build flavors
> 
> Updated all the ci pipelines with **--build_wheel.** This should address this concern. The plan was to do this in another pending PR to avoid merge conflicts. But either way is fine.
> 
> > How did this work before this change?
> 
> x86 builds are not part of CI -- they need to be included, but before we can include them, **run_build.sh** script needs to be fixed so that it doesn't force python tests on x86.

thanks. this makes sense, now that I see there are quite a few CI pipelines not related to an execution provider. the onnxruntime server, C api packaging, and arm cross compile CI's don't need to have python tested. sorry for the back and forth.

File: tools/ci_build/github/azure-pipelines/android-arm64-crosscompile-ci-pipeline.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ jobs:
   steps:
     - template: templates/set-test-data-variables-step.yml
 
-    - script: 'tools/ci_build/github/linux/run_dockerbuild.sh -o android -r $(Build.BinariesDirectory) -d cpu'
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/android-arm64-crosscompile-ci-pipeline.yml]: build_wheel can be left off for this CI

File: tools/ci_build/github/azure-pipelines/android-arm64-crosscompile-ci-pipeline.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ jobs:
...
+    - script: 'tools/ci_build/github/linux/run_dockerbuild.sh -o android -r $(Build.BinariesDirectory) -d cpu -x "--build_wheel"'
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/android-arm64-crosscompile-ci-pipeline.yml]: build_wheel can be left off for arm cross compile I think.

File: tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@ jobs:
     - template: templates/set-test-data-variables-step.yml
     - template: templates/set-version-number-variables-step.yml
 
-    - script: 'tools/ci_build/github/linux/run_dockerbuild.sh -o ubuntu16.04 -d cpu -r $(Build.BinariesDirectory) -x " --test_data_url $(TestDataUrl) --test_data_checksum $(TestDataChecksum)"'
+    - script: 'tools/ci_build/github/linux/run_dockerbuild.sh -o ubuntu16.04 -d cpu -r $(Build.BinariesDirectory) -x "--build_wheel --test_data_url $(TestDataUrl) --test_data_checksum $(TestDataChecksum)"'
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml]: build_wheel can be left off for c api packaging.

File: tools/ci_build/github/azure-pipelines/linux-ort-srv-ci-pipeline.yml [hunk 4]
```diff
@@ -30,7 +30,7 @@ jobs:
         pythonInterpreter: '/usr/bin/python3'
         workingDirectory: $(Build.BinariesDirectory)
 
-    - script: 'tools/ci_build/github/linux/run_dockerbuild.sh -o ubuntu16.04 -d cpu -r $(Build.BinariesDirectory) -x "--config Debug --build_server --use_openmp --use_full_protobuf --enable_server_tests"'
+    - script: 'tools/ci_build/github/linux/run_dockerbuild.sh -o ubuntu16.04 -d cpu -r $(Build.BinariesDirectory) -x "--config Debug --build_server --use_openmp --use_full_protobuf --enable_server_tests --build_wheel"'
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/linux-ort-srv-ci-pipeline.yml]: ort server doesn't need to build python wheel

**Comment by snnn** [comment]
x86 should support python.

File: tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml (hunk 3 above)
Comment (jignparm) [reply tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml]: Done

File: tools/ci_build/github/azure-pipelines/linux-ort-srv-ci-pipeline.yml (hunk 4 above)
Comment (jignparm) [reply tools/ci_build/github/azure-pipelines/linux-ort-srv-ci-pipeline.yml]: Done

File: tools/ci_build/github/azure-pipelines/android-arm64-crosscompile-ci-pipeline.yml (hunk 2 above)
Comment (jignparm) [reply tools/ci_build/github/azure-pipelines/android-arm64-crosscompile-ci-pipeline.yml]: Done

**Pull request merged**